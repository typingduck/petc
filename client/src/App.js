import React from 'react'

import './App.css'

import HomePage from './HomePage'

export default function App () {
  return <HomePage />
}
