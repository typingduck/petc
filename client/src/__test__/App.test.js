/* global expect it jest */
import React from 'react'
import Renderer from 'react-test-renderer'

import App from '../App'

jest.mock('../model/model', () => ({
  initialDoc: () => ({ nodes: {}, edges: {} }),
  loadDoc: () => Promise.resolve({ nodes: {}, edges: {} }),
  connectSocket: () => Promise.resolve()
}))

jest.mock('react-dom', () => ({
  findDOMNode: () => ({
    getBoundingClientRect: () => ({
      left: 10,
      right: 20,
      top: 100,
      bottom: 110
    })
  })
}))

it('renders document page', () => {
  window.history.replaceState(null, null, '/docs/doc-id#node')
  const component = Renderer.create(<App />)
  expect(component.toJSON()).toMatchSnapshot()
})

it('renders homepage', () => {
  window.history.replaceState(null, null, '/')
  const component = Renderer.create(<App />)
  expect(component.toJSON()).toMatchSnapshot()
})
