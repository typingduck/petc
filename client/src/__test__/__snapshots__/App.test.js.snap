// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders document page #edge 1`] = `
<div>
  <div
    id="petc-view-mode-control"
  >
    <ul>
       
      <li
        className=""
      >
        <a
          href="#view"
          id="petc-vmc-view"
          title="view only"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m20 15q2 0 3.5 1.5t1.5 3.5-1.5 3.5-3.5 1.5-3.5-1.5-1.5-3.5 1.5-3.5 3.5-1.5z m0 13.4q3.4 0 5.9-2.5t2.5-5.9-2.5-5.9-5.9-2.5-5.9 2.5-2.5 5.9 2.5 5.9 5.9 2.5z m0-20.9q6.2 0 11.2 3.4t7.2 9.1q-2.2 5.6-7.2 9.1t-11.2 3.4-11.2-3.4-7.2-9.1q2.2-5.6 7.2-9.1t11.2-3.4z"
              />
            </g>
          </svg>
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#node"
          id="petc-vmc-node"
          title="edit nodes"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m28.4 26.6h6.6v8.4h-8.4v-5.1l-6.6-7-6.6 7v5.1h-8.4v-8.4h6.6l6.8-6.6v-5.3c-2-0.7-3.4-2.5-3.4-4.7 0-2.7 2.3-5 5-5s5 2.3 5 5c0 2.2-1.4 4-3.4 4.7v5.3z"
              />
            </g>
          </svg>
        </a>
      </li>
      <li
        className="active"
      >
        <a
          href="#edge"
          id="petc-vmc-edge"
          title="edit edges"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m24.7 22.3l5.2 5.3 3.5-3.5v9.3h-9.3l3.5-3.5-5.3-5.2z m-0.6-15.7h9.3v9.3l-3.5-3.5-20.9 21-2.4-2.4 21-20.9z m-6.4 8.7l-2.4 2.4-8.7-8.7 2.4-2.4z"
              />
            </g>
          </svg>
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="/"
          id="petc-vmc-demo"
          title="home"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m31.6 23.4v-3.4h-5v-5h-3.2v5h-5v3.4h5v5h3.2v-5h5z m1.8-13.4q1.4 0 2.3 1t0.9 2.4v16.6q0 1.4-0.9 2.4t-2.3 1h-26.8q-1.4 0-2.3-1t-0.9-2.4v-20q0-1.4 0.9-2.4t2.3-1h10l3.4 3.4h13.4z"
              />
            </g>
          </svg>
        </a>
      </li>
       
    </ul>
  </div>
  <div
    className="petc-pageview"
  >
    <h1>
      pannus et circulos
    </h1>
    <div
      id="petc-kanvas"
      onClick={[Function]}
    />
    <div
      className="hidden"
      id="petc-trashcan"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m31.6 6.6v3.4h-23.2v-3.4h5.7l1.8-1.6h8.2l1.8 1.6h5.7z m-21.6 25v-20h20v20c0 1.8-1.6 3.4-3.4 3.4h-13.2c-1.8 0-3.4-1.6-3.4-3.4z"
          />
        </g>
      </svg>
    </div>
    <div
      id="petc-edge-select-control"
    >
      <ul>
        <li
          className={null}
          onClick={[Function]}
        >
          <div
            className="petc-edge-select-element"
          >
            <div
              className="petc-edge-select-node-a"
              id="petc-edge-select-element-default-a"
            />
            <div
              className="petc-edge-select-node-b"
              id="petc-edge-select-element-default-b"
            />
          </div>
          <div>
            default
          </div>
        </li>
        <li
          onClick={[Function]}
        >
           + new 
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`renders document page #node 1`] = `
<div>
  <div
    id="petc-view-mode-control"
  >
    <ul>
       
      <li
        className=""
      >
        <a
          href="#view"
          id="petc-vmc-view"
          title="view only"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m20 15q2 0 3.5 1.5t1.5 3.5-1.5 3.5-3.5 1.5-3.5-1.5-1.5-3.5 1.5-3.5 3.5-1.5z m0 13.4q3.4 0 5.9-2.5t2.5-5.9-2.5-5.9-5.9-2.5-5.9 2.5-2.5 5.9 2.5 5.9 5.9 2.5z m0-20.9q6.2 0 11.2 3.4t7.2 9.1q-2.2 5.6-7.2 9.1t-11.2 3.4-11.2-3.4-7.2-9.1q2.2-5.6 7.2-9.1t11.2-3.4z"
              />
            </g>
          </svg>
        </a>
      </li>
      <li
        className="active"
      >
        <a
          href="#node"
          id="petc-vmc-node"
          title="edit nodes"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m28.4 26.6h6.6v8.4h-8.4v-5.1l-6.6-7-6.6 7v5.1h-8.4v-8.4h6.6l6.8-6.6v-5.3c-2-0.7-3.4-2.5-3.4-4.7 0-2.7 2.3-5 5-5s5 2.3 5 5c0 2.2-1.4 4-3.4 4.7v5.3z"
              />
            </g>
          </svg>
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#edge"
          id="petc-vmc-edge"
          title="edit edges"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m24.7 22.3l5.2 5.3 3.5-3.5v9.3h-9.3l3.5-3.5-5.3-5.2z m-0.6-15.7h9.3v9.3l-3.5-3.5-20.9 21-2.4-2.4 21-20.9z m-6.4 8.7l-2.4 2.4-8.7-8.7 2.4-2.4z"
              />
            </g>
          </svg>
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="/"
          id="petc-vmc-demo"
          title="home"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m31.6 23.4v-3.4h-5v-5h-3.2v5h-5v3.4h5v5h3.2v-5h5z m1.8-13.4q1.4 0 2.3 1t0.9 2.4v16.6q0 1.4-0.9 2.4t-2.3 1h-26.8q-1.4 0-2.3-1t-0.9-2.4v-20q0-1.4 0.9-2.4t2.3-1h10l3.4 3.4h13.4z"
              />
            </g>
          </svg>
        </a>
      </li>
       
    </ul>
  </div>
  <div
    className="petc-pageview"
  >
    <h1>
      pannus et circulos
    </h1>
    <div
      id="petc-kanvas"
      onClick={[Function]}
    />
    <div
      className=""
      id="petc-trashcan"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m31.6 6.6v3.4h-23.2v-3.4h5.7l1.8-1.6h8.2l1.8 1.6h5.7z m-21.6 25v-20h20v20c0 1.8-1.6 3.4-3.4 3.4h-13.2c-1.8 0-3.4-1.6-3.4-3.4z"
          />
        </g>
      </svg>
    </div>
    <div
      id="petc-node-select-control"
    >
      <ul>
        <li
          className={null}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
        >
          <div
            className="petc-node default react-draggable"
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "touchAction": "none",
                "transform": "translate(0px,0px)",
              }
            }
            transform={null}
          />
          <div>
            default
          </div>
        </li>
        <li
          onClick={[Function]}
        >
           + new 
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`renders document page 1`] = `
<div>
  <div
    id="petc-view-mode-control"
  >
    <ul>
       
      <li
        className="active"
      >
        <a
          href="#view"
          id="petc-vmc-view"
          title="view only"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m20 15q2 0 3.5 1.5t1.5 3.5-1.5 3.5-3.5 1.5-3.5-1.5-1.5-3.5 1.5-3.5 3.5-1.5z m0 13.4q3.4 0 5.9-2.5t2.5-5.9-2.5-5.9-5.9-2.5-5.9 2.5-2.5 5.9 2.5 5.9 5.9 2.5z m0-20.9q6.2 0 11.2 3.4t7.2 9.1q-2.2 5.6-7.2 9.1t-11.2 3.4-11.2-3.4-7.2-9.1q2.2-5.6 7.2-9.1t11.2-3.4z"
              />
            </g>
          </svg>
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#node"
          id="petc-vmc-node"
          title="edit nodes"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m28.4 26.6h6.6v8.4h-8.4v-5.1l-6.6-7-6.6 7v5.1h-8.4v-8.4h6.6l6.8-6.6v-5.3c-2-0.7-3.4-2.5-3.4-4.7 0-2.7 2.3-5 5-5s5 2.3 5 5c0 2.2-1.4 4-3.4 4.7v5.3z"
              />
            </g>
          </svg>
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#edge"
          id="petc-vmc-edge"
          title="edit edges"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m24.7 22.3l5.2 5.3 3.5-3.5v9.3h-9.3l3.5-3.5-5.3-5.2z m-0.6-15.7h9.3v9.3l-3.5-3.5-20.9 21-2.4-2.4 21-20.9z m-6.4 8.7l-2.4 2.4-8.7-8.7 2.4-2.4z"
              />
            </g>
          </svg>
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="/"
          id="petc-vmc-demo"
          title="home"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m31.6 23.4v-3.4h-5v-5h-3.2v5h-5v3.4h5v5h3.2v-5h5z m1.8-13.4q1.4 0 2.3 1t0.9 2.4v16.6q0 1.4-0.9 2.4t-2.3 1h-26.8q-1.4 0-2.3-1t-0.9-2.4v-20q0-1.4 0.9-2.4t2.3-1h10l3.4 3.4h13.4z"
              />
            </g>
          </svg>
        </a>
      </li>
       
    </ul>
  </div>
  <div
    className="petc-pageview"
  >
    <h1>
      pannus et circulos
    </h1>
    <div
      id="petc-kanvas"
      onClick={[Function]}
    />
    <div
      className="hidden"
      id="petc-trashcan"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m31.6 6.6v3.4h-23.2v-3.4h5.7l1.8-1.6h8.2l1.8 1.6h5.7z m-21.6 25v-20h20v20c0 1.8-1.6 3.4-3.4 3.4h-13.2c-1.8 0-3.4-1.6-3.4-3.4z"
          />
        </g>
      </svg>
    </div>
  </div>
</div>
`;

exports[`renders homepage 1`] = `
<div
  className="petc-pageview"
>
  <h1>
    pannus et circulos
  </h1>
  <div
    className="petc-homepage"
  >
    <a
      href="#new"
      id="petc-create-new"
      onClick={[Function]}
    >
      Create New
    </a>
    <br />
    <a
      href="#demo"
      id="petc-create-demo"
      onClick={[Function]}
    >
      Demo doc
    </a>
  </div>
</div>
`;
